crates/flipperzero/src/furi.rs

```diff
@@ -38,3 +40,60 @@ pub fn sleep(duration: core::time::Duration) {
         }
     }
 }
+
+/// MessageQueue provides a safe wrapper around the furi message queue primitive.
+pub struct MessageQueue<M: Sized> {
+    hnd: *const sys::message_queue::FuriMessageQueue,
+    _marker: core::marker::PhantomData<M>,
+}
+
+impl<M: Sized> MessageQueue<M> {
+    /// Constructs a message queue with the given capacity.
+    pub fn new(capacity: u32) -> Self {
+        Self {
+            hnd: unsafe { sys::message_queue::alloc(capacity, core::mem::size_of::<M>()) },
+            _marker: core::marker::PhantomData::<M>,
+        }
+    }
+
+    // Attempts to add the message to the end of the queue, waiting up to timeout ticks.
+    // FuriStatusOk (0) is returned if the insert is successful, otherwise FuriStatusErrorTimeout
+    // (-2) is returned.
+    pub fn put(&self, mut msg: M, timeout: u32) -> u32 {
```

Comment:
For the high-level APIs, we should return some sort of `Result<T, Error>` rather than the error code directly.

Something probably a bit like [`nix::Error`](https://docs.rs/nix/latest/nix/type.Error.html).